### ISSUE TITLE
cdb2jdbc does not build

### ISSUE THREAD

**Issue by dennisgove** [bloomberg/comdb2#21]
I tried to build the jdbc driver with maven but it failed to build due to what looks like a dependency issue on protobuf

Full stacktrace can be found at https://gist.github.com/dennisgove/829fa028237d9b2a9f967d4152bee8ce

Pertinent parts are

```
~/dev/comdb2/cdb2jdbc $ mvn clean install
[INFO] Scanning for projects...
[INFO]
[INFO] Using the builder org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder with a thread count of 1
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Building cdb2jdbc 2.0.0
[INFO] ------------------------------------------------------------------------
Downloading: http://repo.maven.apache.org/maven2/org/xolstice/maven/plugins/protobuf-maven-plugin/0.5.0/protobuf-maven-plugin-0.5.0.pom
.
.
[WARNING] No 'protocExecutable' parameter is configured, using the default: 'protoc'
[INFO] Compiling 1 proto file(s) to /Users/dennis/dev/comdb2/cdb2jdbc/target/generated-sources/sqlquery
[INFO]
[INFO] --- protobuf-maven-plugin:0.5.0:compile (compile_sqlresponse) @ cdb2jdbc ---
[WARNING] No 'protocExecutable' parameter is configured, using the default: 'protoc'
[INFO] Compiling 1 proto file(s) to /Users/dennis/dev/comdb2/cdb2jdbc/target/generated-sources/sqlresponse
[INFO]
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ cdb2jdbc ---
[WARNING] Using platform encoding (UTF-8 actually) to copy filtered resources, i.e. build is platform dependent!
[INFO] skip non existing resourceDirectory /Users/dennis/dev/comdb2/cdb2jdbc/src/main/resources
[INFO] Copying 1 resource
[INFO] Copying 1 resource
[INFO]
[INFO] --- maven-compiler-plugin:2.5.1:compile (default-compile) @ cdb2jdbc ---
[WARNING] File encoding has not been set, using platform encoding UTF-8, i.e. build is platform dependent!
[INFO] Compiling 34 source files to /Users/dennis/dev/comdb2/cdb2jdbc/target/classes
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR :
[INFO] -------------------------------------------------------------
[ERROR] /Users/dennis/dev/comdb2/cdb2jdbc/target/generated-sources/sqlquery/com/bloomberg/comdb2/jdbc/Sqlquery.java:[7863,42] error: package com.google.protobuf.GeneratedMessageV3 does not exist
[ERROR] /Users/dennis/dev/comdb2/cdb2jdbc/target/generated-sources/sqlquery/com/bloomberg/comdb2/jdbc/Sqlquery.java:[7868,42] error: package com.google.protobuf.GeneratedMessageV3 does not exist
[ERROR] /Users/dennis/dev/comdb2/cdb2jdbc/target/generated-sources/sqlquery/com/bloomberg/comdb2/jdbc/Sqlquery.java:[7873,42] error: package com.google.protobuf.GeneratedMessageV3 does not exist
[ERROR] /Users/dennis/dev/comdb2/cdb2jdbc/target/generated-sources/sqlquery/com/bloomberg/comdb2/jdbc/Sqlquery.java:[7878,42] error: package com.google.protobuf.GeneratedMessageV3 does not exist
[ERROR] /Users/dennis/dev/comdb2/cdb2jdbc/target/generated-sources/sqlquery/com/bloomberg/comdb2/jdbc/Sqlquery.java:[7883,42] error: package com.google.protobuf.GeneratedMessageV3 does not exist
[ERROR] /Users/dennis/dev/comdb2/cdb2jdbc/target/generated-sources/sqlquery/com/bloomberg/comdb2/jdbc/Sqlquery.java:[7888,42] error: package com.google.protobuf.GeneratedMessageV3 does not exist
[ERROR] /Users/dennis/dev/comdb2/cdb2jdbc/target/generated-sources/sqlquery/com/bloomberg/comdb2/jdbc/Sqlquery.java:[7893,42] error: package com.google.protobuf.GeneratedMessageV3 does not exist
[ERROR] /Users/dennis/dev/comdb2/cdb2jdbc/target/generated-sources/sqlquery/com/bloomberg/comdb2/jdbc/Sqlquery.java:[293,25] error: cannot find symbol
[ERROR]   symbol:   class GeneratedMessageV3
  location: package com.google.protobuf
/Users/dennis/dev/comdb2/cdb2jdbc/target/generated-sources/sqlquery/com/bloomberg/comdb2/jdbc/Sqlquery.java:[297,60] error: package com.google.protobuf.GeneratedMessageV3 does not exist
.
.
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

**Comment by dennisgove** [comment]
My version of `protoc` is 3.2.0
```
$ protoc --version
libprotoc 3.2.0
```

**Comment by riverszhang89** [comment]
Hi Dennis,

```
[WARNING] No 'protocExecutable' parameter is configured, using the default: 'protoc'
```
What is the version of the protobuf compiler installed on your machine? cdb2jdbc uses protobuf-java 2.6.1 by default. This could be some compatibility issue between the library and compiler.

**Comment by riverszhang89** [comment]
You could downgrade the compiler to 2.6.1, or change the version in pom.xml to 3.2.0

**Comment by dennisgove** [comment]
Good call. The following diff allows the build to succeed.
```
$ git diff
diff --git a/cdb2jdbc/pom.xml b/cdb2jdbc/pom.xml
index abea573..49e2710 100644
--- a/cdb2jdbc/pom.xml
+++ b/cdb2jdbc/pom.xml
@@ -26,7 +26,7 @@ limitations under the License. -->
     <dependency>
       <groupId>com.google.protobuf</groupId>
       <artifactId>protobuf-java</artifactId>
-      <version>2.6.1</version>
+      <version>3.2.0</version>
     </dependency>
     <dependency>
         <groupId>log4j</groupId>
```

**Comment by riverszhang89** [comment]
:+1: Thanks for the resolution.

**Comment by mponomar** [comment]
Fixed by #31 (Thanks Dennis!).  One minor quibble is that building the driver in docker places root-owned files in cdb2jdbc/target, and subsequently needs sudo mvn clean instead of mvn clean.  That's a matter for the docs though, as the Docker build is very useful.

**Comment by dennisgove** [comment]
Good catch. That's actually easy to fix. The container itself can be run as the user and then resulting artifacts will have the normal ownership settings. I'll make that change.
